wrappers/python/simtk/openmm/app/amberprmtopfile.py

```diff
@@ -248,4 +254,16 @@ def createSystem(self, nonbondedMethod=ff.NoCutoff, nonbondedCutoff=1.0*unit.nan
                 force.setEwaldErrorTolerance(ewaldErrorTolerance)
         if removeCMMotion:
             sys.addForce(mm.CMMotionRemover())
+
+        if switchDistance and nonbondedMethod is not ff.NoCutoff:
+            # make sure it's legal
+            if switchDistance >= nonbondedCutoff:
+                raise ValueError('switchDistance is too large compared '
+                                     'to the cutoff!')
+                if abs(switchDistance) != switchDistance:
```

Comment:
Can do. That was a copy and paste job from the CharmmPsfFile code
On 8 Oct 2015 20:07, "Jason Swails" <notifications@github.com> wrote:

> In wrappers/python/simtk/openmm/app/amberprmtopfile.py
> <https://github.com/pandegroup/openmm/pull/1177#discussion_r41547062>:
>
> > @@ -248,4 +254,16 @@ def createSystem(self, nonbondedMethod=ff.NoCutoff, nonbondedCutoff=1.0*unit.nan
> >                  force.setEwaldErrorTolerance(ewaldErrorTolerance)
> >          if removeCMMotion:
> >              sys.addForce(mm.CMMotionRemover())
> > +
> > +        if switchDistance and nonbondedMethod is not ff.NoCutoff:
> > +            # make sure it's legal
> > +            if switchDistance >= nonbondedCutoff:
> > +                raise ValueError('switchDistance is too large compared '
> > +                                     'to the cutoff!')
> > +                if abs(switchDistance) != switchDistance:
>
> If you do the above, this can be simplified to switchDistance < 0, which
> will be easier to see intent immediately.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pandegroup/openmm/pull/1177/files#r41547062>.
>